src/classes/BDI_ContactService.cls

```diff
@@ -894,6 +893,7 @@ public with sharing class BDI_ContactService {
                 diKey += String.valueOf(di.get(strCx + '_Alternate_Email__c'));
             }
             diKey += '|';
+            diKey.toLowerCase();
```

Comment:
Let me write a better test, I believe this is needed.

Reply:
This is what I ran in exec anon to confirm that this wouldn't work:

```
String something = 'Foo';
System.debug(something); // prints Foo
something.toLowerCase();
System.debug(something); // prints Foo
something = something.toLowerCase();
System.debug(something); // prints foo
```

Reply:
I commented out this entire block (lines 885-914) and no tests failed because of it. I'm wary of removing it, though, so I'll just fix the lowercasing and leave it be.